x-pack/platform/plugins/private/observability_ai_assistant_management/public/routes/components/settings_tab/change_kb_model.tsx

```diff
@@ -26,12 +26,13 @@ import {
 } from '@kbn/ai-assistant/src/utils/get_model_options_for_inference_endpoints';
 import { useInferenceEndpoints, UseKnowledgeBaseResult } from '@kbn/ai-assistant/src/hooks';
 import { KnowledgeBaseState, useKibana } from '@kbn/observability-ai-assistant-plugin/public';
+import { useInferenceId } from './inference_id_context';
 
 export function ChangeKbModel({ knowledgeBase }: { knowledgeBase: UseKnowledgeBaseResult }) {
   const { overlays } = useKibana().services;
+  const { selectedInferenceId, setSelectedInferenceId } = useInferenceId();
```

Comment:
Do we need `useInferenceId`? Can't we get the current inference id from `knowledgeBase`: `knowledgeBase.status.value.currentInferenceId` ?